apps/web/composables/useNotification/__tests__/useNotification.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { useNotification } from '../useNotification';
+
+describe('useNotification', () => {
+
+    afterEach(() => {
+        const { clear } = useNotification();
+        clear();
+    });
+
+    it('should return the correct initial state', () => {
+        const { data, send } = useNotification();
+
+        expect(data.value).toEqual([]);
+        expect(send).toBeInstanceOf(Function);
+    });
+
+    it('should add a notification', () => {
+        const { data, send } = useNotification();
+
+        send({
+            message: 'Test alert error with a longer message',
+            type: 'negative',
+            persist: true,
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+        });
+
+        expect(data.value.length).toBe(1);
+    });
+
+    it('should remove a notification', () => {
+        const { data, send } = useNotification();
+
+        send({
+            message: 'Test alert error with a longer message',
+            type: 'negative',
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+        });
+
+        expect(data.value.length).toBe(1);
+
+        if (data.value[0].dismiss) {
+            data.value[0].dismiss();
+        }
+
+        expect(data.value.length).toBe(0);
+    });
+
+    it('should clear all notifications', () => {
+        const { data, send, clear } = useNotification();
+
+        send({
+            message: 'Test alert error with a longer message',
+            type: 'negative',
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+        });
+
+        expect(data.value.length).toBe(1);
+
+        clear();
+
+        expect(data.value.length).toBe(0);
+    });
+
+    it('should remove the oldest notification when the maximum number of notifications is reached', async () => {
+        const { data, send } = useNotification();
+
+        for (let i = 0; i < 6; i++) {
+            send({
+                message: 'Test positive message',
+                type: 'positive',
+                action: {
+                    text: 'action',
+                    onClick: () => {},
+                },
+            });
+        }
+
+        expect(data.value.length).toBe(5);
+    });
+
+    it('should remove a notification after 10ms', async () => {
+        const { data, send } = useNotification();
+
+        send({
+            message: 'Test positive message',
+            type: 'positive',
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+            dismissTimeout: 10
+        });
+
+        expect(data.value.length).toBe(1);
+
+        await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
**suggestion**: Add a threshold test. That is, set a timeout for 9 seconds and check if `length` is still `1`.

Reply:
done, it now tests the default timeout of 3000ms and checks right before the end if the message is still there